auth/saml/index.php

```diff
@@ -9,11 +9,25 @@
         // We read saml parameters from a config file instead from the database
         // due we can not operate with the moodle database without load all
         // moodle session issue.
-        if(!file_exists('saml_config.php')) {
+				//We need to get dataroot from CFG without loading moodle database
+
+				$config = file_get_contents('../../config.php');
+
+				$config = preg_replace('/\s+/', '', $config);
+				if(preg_match('/^.+?\$CFG\-\>dataroot=[\'"](.+?)[\'"];/', $config, $match)){
+					$CFG->dataroot = $match[1];
+				}else {
+            throw(new Exception('Moodle dataroot not found'));
+				}
+				
+				die(var_dump($CFG));
+ 
+        if(!file_exists($CFG->dataroot.'/saml_config.php')) {
```

Comment:
The file is actually included in git, which should make it impossible to not have the file. However, it should be created during the upgrade/install-process. I can create an upgrade path from a specified version before the file was included, which checks for a present file in the plugin-dir and the copies it to the moodledata-dir, if it has been changed. Agree?